File: WalletService.py
from AccountService import AccountService
from MoneyUtils import MoneyUtils


class WalletService:
    @staticmethod
    def check_balance(user):
        print("Wallet Balance:", user.wallet.balance)

    @staticmethod
    def add_funds(user):
        print("Enter the amount (in MGA) to add:")
        amount = float(input())
        if MoneyUtils.is_amount_valid(amount):
            user.wallet.deposit(amount)
            print("Funds added successfully. New balance:", user.wallet.balance, "MGA")

    @staticmethod
    def withdraw_money(user):
        print("Enter the amount (in MGA) to withdraw:")
        amount = float(input())
        if user.wallet.withdraw(amount):
            print("Withdrawal successful. New balance:", user.wallet.balance, "MGA")

    @staticmethod
    def peer_to_peer_transfers(user, users):
        print("Enter the destination username:")
        destination_username = input()

        destination_user = AccountService.find_user_by_username(destination_username, users)

        if destination_user:
            print("Enter the amount (in MGA) to transfer:")
            amount = float(input())
            if user.wallet.transfer(amount, destination_user):
                print(amount, "MGA sent to:", destination_username)
                print("Transfer successful. New balance:", user.wallet.balance, "MGA")
        else:
            print("User not found with the provided username.")

    @staticmethod
    def transaction_history(user):
        transactions = user.wallet.transaction_history.transactions

        print("Transaction History:")
        for transaction in transactions:
            print("Transaction ID:", transaction.id)
            print("Amount:", transaction.amount, "MGA")
            print("Type:", transaction.type)
            print("Destination:", transaction.destination)
            print("Transaction Date:", transaction.transaction_date)
            print("------")

File: WalletService.js
class WalletService {
    static checkBalance(user) {
        console.log("Wallet Balance:", user.wallet.balance);
    }

    static addFunds(user) {
        console.log("Enter the amount (in MGA) to add:");
        const amount = parseFloat(prompt());

        if (MoneyUtils.isAmountValid(amount)) {
            user.wallet.deposit(amount);
            console.log("Funds added successfully. New balance:", user.wallet.balance, "MGA");
        }
    }

    static withdrawMoney(user) {
        console.log("Enter the amount (in MGA) to withdraw:");
        const amount = parseFloat(prompt());

        if (user.wallet.withdraw(amount)) {
            console.log("Withdrawal successful. New balance:", user.wallet.balance, "MGA");
        }
    }

    static peerToPeerTransfers(user, users) {
        console.log("Enter the destination username:");
        const destinationUsername = prompt();

        const destinationUser = AccountService.findUserByUsername(destinationUsername, users);

        if (destinationUser) {
            console.log("Enter the amount (in MGA) to transfer:");
            const amount = parseFloat(prompt());

            if (user.wallet.transfer(amount, destinationUser)) {
                console.log(amount, "MGA sent to:", destinationUsername);
                console.log("Transfer successful. New balance:", user.wallet.balance, "MGA");
            }
        } else {
            console.log("User not found with the provided username.");
        }
    }

    static transactionHistory(user) {
        const transactions = user.wallet.transactionHistory.transactions;

        console.log("Transaction History:");
        transactions.forEach(transaction => {
            console.log("Transaction ID:", transaction.id);
            console.log("Amount:", transaction.amount, "MGA");
            console.log("Type:", transaction.type);
            console.log("Destination:", transaction.destination);
            console.log("Transaction Date:", transaction.transactionDate);
            console.log("------");
        });
    }
}

module.exports = { WalletService };
